File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.java
import java.util.*;

class Solution {
    public int solution(int n) {
        
        if (n == 2) return 1;
        
        int[] ans = new int[n+1];
        ans[1] = ans[2] = 1;
        
        for (int i = 3; i <= n; i++) ans[i] = (ans[i-2] + ans[i-1]) % 1234567;
        
        return ans[n];
        
    }
}

File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.js
function solution(n) {
    if (n === 2) return 1;
    
    let arr = [];
    arr[0] = 0;
    arr[1] = 1;
    arr[2] = 1;
    
    for (let i = 3; i <= n; i++) {
        arr[i] = (arr[i - 2] + arr[i - 1]) % 1234567;
    }
    
    return arr[n];
}